implementations/rust/ockam/ockam/src/forwarder.rs

```diff
@@ -0,0 +1,96 @@
+use crate::Context;
+use ockam_core::{Address, Any, LocalMessage, Result, Route, Routed, TransportMessage, Worker};
+use tracing::info;
+
+/// Alias worker to register remote workers under local names.
+///
+/// To talk with this worker, you can use the [`RemoteForwarder`](crate::RemoteForwarder)
+/// which is a compatible client for this server.
+#[non_exhaustive]
+pub struct ForwardingService {}
+
+impl ForwardingService {
+    /// Start a forwarding service. The address of the forwarding service will be
+    /// `forwarding_service`.
+    pub async fn create(ctx: &Context) -> Result<()> {
+        ctx.start_worker("forwarding_service", Self {}).await?;
+        Ok(())
+    }
+}
+
+#[crate::worker]
+impl Worker for ForwardingService {
+    type Context = Context;
+    type Message = Any;
+
+    async fn handle_message(
+        &mut self,
+        ctx: &mut Self::Context,
+        msg: Routed<Self::Message>,
+    ) -> Result<()> {
+        let forward_route = msg.return_route();
+        let payload = msg.into_transport_message().payload;
+        Forwarder::create(ctx, forward_route, payload).await?;
+
+        Ok(())
+    }
+}
+
+struct Forwarder {
+    forward_route: Route,
+    payload: Option<Vec<u8>>,
```

Comment:
I think a comment about why this is `Option`/when this is `None` might be worth it.

It's so that this extra information can be passed from the `create` function into `initialize`, right?

(I think this is probably a bit of an API wart with Workers work that we' like to address eventually, but for now just a comment would help)

Reply:
> It's so that this extra information can be passed from the create function into initialize, right?

Exactly.